lib/services/devices/deviceRegistryMemory.js

```diff
@@ -110,16 +128,20 @@ function listDevices(service, subservice, limit, offset, callback) {
     });
 }
 
-function getDevice(id, callback) {
-    if (registeredDevices[id]) {
-        callback(null, registeredDevices[id]);
+function getDevice(id, service, subservice, callback) {
+    if (!subservice && !callback) {
+        callback = service;
+    }
+
+    if (registeredDevices[service] && registeredDevices[service][id]) {
```

Comment:
You are right... That shouldn't be the case, actually. One of the overall features of this PR is to make the service and subservice used along the full stack, so there shouldn't be any calls without service. I have filed an issue to add checks for all the internal methods, [here](https://github.com/telefonicaid/iotagent-node-lib/issues/349). Requests from the Provisioning API are required to have the fiware-XXX headers.

All that said, I'll remove the guards against service and subservice in the registry methods.